basis/math/matrices/matrices.factor

```diff
@@ -6,25 +6,49 @@ sequences.private fry math.statistics grouping
 combinators.short-circuit math.ranges combinators.smart ;
 IN: math.matrices
 
+DEFER: eye
+<PRIVATE
+: keep-order ( ..a x quot: ( ..a x -- ..b ) -- ..b x )
+    [ dup ] dip call ; inline
+
+PRIVATE>
+
+ERROR: negative-power-matrix m n ;
+
 ! Matrices
-: make-matrix ( m n quot -- matrix )
+: make-matrix ( m n quot: ( ... -- elt ) -- matrix )
     '[ _ _ replicate ] replicate ; inline
 
+: make-matrix-with-indices ( m n quot -- matrix )
+    [ [ <iota> ] bi@ ] dip '[ @ ] cartesian-map ; inline
```

Comment:
can't ``'[ @ ]`` be skipped?

```factor
: make-matrix-with-indices ( m n quot -- matrix )
    [ [ <iota> ] bi@ ] dip cartesian-map ; inline
```

Reply:
Also, maybe put the stack effects ``( ... m n quot: ( ... m' n' -- ... elt ) -- ... matrix )``